docker/pytorch/Dockerfile

```diff
@@ -102,11 +106,30 @@ RUN pip install --no-cache-dir --find-links https://download.pytorch.org/whl/tor
 #########
 RUN rm -rf /tmp/*
 
+
+#######################
+# Set the shell to bash
+#######################
+SHELL ["/bin/bash", "-c"]
+
+####################################################
+# Create a default virtualenv and add it to the path
+####################################################
+RUN python${PYTHON_VERSION} -m venv --system-site-packages /mosaicml-venv && \
+    /mosaicml-venv/bin/python -m pip install --upgrade pip setuptools && \
+    # Set the path for non-interactive shells
+    sed -i 's/PATH=\"/PATH=\"~\/.local\/bin:\/mosaicml-venv\/bin:\/usr\/local\/nvidia\/bin:\/usr\/local\/cuda\/bin:/g' /etc/environment && \
+    echo "export PATH=~/.local/bin:/mosaicml-venv/bin:$PATH" >> /etc/profile && \
+    # And for interactive shells
+    echo "export PATH=~/.local/bin:/mosaicml-venv/bin:$PATH" >> /etc/bash.bashrc
```

Comment:
Not completely sure if this is needed but I think we also need to set the `VIRTUAL_ENV` environment variable to be safe as some packages use it to check if they're running within an environment.  Found the following article on this, see the end but it says set `VIRTUAL_ENV` and then add the `$VIRTUAL_ENV/bin` path to `$PATH`:
https://pythonspeed.com/articles/activate-virtualenv-dockerfile/